sstables/sstables.cc

```diff
@@ -142,8 +142,14 @@ future<file> sstable::new_sstable_component_file(const io_error_handler& error_h
         return make_checked_file(error_handler, std::move(f));
     });
 
-    return f.handle_exception([this, type] (auto ep) {
+    return f.handle_exception([this, type, &error_handler] (auto ep) {
         sstlog.error("Could not create SSTable component {}. Found exception: {}", filename(type), ep);
+        try {
+            error_handler(ep);
+        } catch (...) {
+            ep = std::current_exception();
+        }
+        std::rethrow_exception(ep);
```

Comment:
`std::move(ep)`

Reply:
Also, how is the above try-catch-rethrow different from just not catching whatever `error_handler()` throws?

Reply:
It is not. But it is not supposed to be there. Sorry, copy-paste from playing with coroutinization. There is a return statement after...